src/ngraph/runtime/cpu/pass/rnn_fusion.cpp

```diff
@@ -0,0 +1,566 @@
+                                             num_fused_rnn_layers);
+
+        std::vector<std::shared_ptr<op::Slice>> ht_slice_per_timestep(num_of_lstm_matched, nullptr);
+        auto rnn_ht_out = std::make_shared<op::GetOutputElement>(rnn, 0);
+        auto rnn_ct_out = std::make_shared<op::GetOutputElement>(rnn, 1);
+
+        //slice the rnn ht's
+        size_t start_index = 0;
+        size_t end_index = batch_size;
+        // capture the slices in the reverse order, so it corrosponds to lstm_goes order captured by the Pattern matcher
+        for (size_t i = 0; i < num_of_lstm_matched; i++)
+        {
+            ht_slice_per_timestep[i] = (std::make_shared<op::Slice>(
+                rnn_ht_out, Coordinate{start_index, 0}, Coordinate{end_index, feature_size}));
+            start_index += batch_size;
+            end_index += batch_size;
+        }
+        std::reverse(ht_slice_per_timestep.begin(), ht_slice_per_timestep.end());
+
+        NGRAPH_DEBUG << "rnn_time_slice: " << ht_slice_per_timestep.size();
+
+        // find the lstm's nodes captured in PM
+        auto lstm_goes = m.get_bound_nodes_for_pattern(lstm_node_label);
+        std::vector<std::shared_ptr<ngraph::Node>> lstm_nodes;
+
+        // we need to collect LSTM from GOE's, in order to deterministicaly determine
+        // the individaual time slice output ht. lstm_goes will hold the GOE in the decreasing
+        // order of the time slices
+        for (size_t i = 0; i < lstm_goes.size(); i++)
+        {
+            // lstm's will be the input to GOE's
+            lstm_nodes.push_back(lstm_goes[i]->get_arguments()[0]);
+        }
+
+        if (sequence_len != lstm_nodes.size())
+        {
+            throw ngraph_error(" Number of lstm nodes in RNN layer is not equal to time slices");
+        }
+
+        if (lstm_nodes.size() != lstm_goes.size() &&
+            lstm_goes.size() != ht_slice_per_timestep.size())
+        {
+            throw ngraph_error(
+                "Number of slices of rnn output ht is not equal to the time slices in RNN layer");
+        }
+
+        // collect all the consumers of LSTM goe's (ht)
+        std::set<std::shared_ptr<ngraph::Node>> lstm_goe0_user;
+        std::unordered_map<std::shared_ptr<Node>, std::shared_ptr<Node>> map_goe_to_lstm_slices;
+        std::shared_ptr<Node> goe_0;
+        for (size_t index = 0; index < lstm_nodes.size(); index++)
+        {
+            // now get the GOE0 which is the first output of lstm (ht)
+            for (auto& goes : lstm_nodes[index]->get_outputs().at(0).get_inputs())
+            {
+                auto goe_node = std::dynamic_pointer_cast<op::GetOutputElement>(goes->get_node());
+                // first output node of lstm
+                if (goe_node->get_n() == 0)
+                {
+                    goe_0 = goes->get_node();
+                }
+            }
+
+            for (auto goe0_user : goe_0->get_users())
```

Comment:
this skips intermediate connections  we don't care about and graphs that LSTMs were fused from?